### ISSUE TITLE
[Performance] Replace instances of `process && process.env.NODE_ENV`

### ISSUE THREAD

**Issue by TheLarkInn** [OfficeDev/office-ui-fabric-react#8859]
### Bug/Feature Report
Remove any standalone instances of `process && process.env.NODE_ENV` from source code.

### Backstory
For those who don't know I'm Sean from the [webpack team](https//webpack.js.org) 👋👋👋👋👋👋👋 

Unfortunately, many developers are **unaware** of some of the node/isomorphic polyfilling that we do automatically in webpack. In hindsight, this was for backwards compatibility reasons and allowed for easier isomorphic module development. 

Going forward in webpack 5 we will turn these features off! (Not as relevant). 

Today, webpack looks at a variety of expressions and polyfills them auto-magically. Here is the [list of default](https://webpack.js.org/configuration/node#node) ones we polyfill in v3 and up:

```
module.exports = {
  //...
  node: {
    console: false,
    global: true,
    process: true,
    __filename: 'mock',
    __dirname: 'mock',
    Buffer: true,
    setImmediate: true

    // See "Other node core libraries" for additional options.
  }
};
```

The way that we polyfill, however is done through a process called "variable injection". (If you are webpack-savvy and used the Provide plugin ever [not recommended], then this will be familiar).

The problem: variable injection (according to our [documetnation](https://webpack.js.org/plugins/module-concatenation-plugin/#optimization-bailouts)) is an **Optimization bailout for Scope hoisting (and treeshaking)**. 

**Therefore, any component which uses the above expressions: (`process`, `global`, setImmediate, etc) will not be able to be tree-shaken.**

Note: you can leave `process.env.NODE_ENV` because webpack during compile replaces this with a string. Therefore it won't be subject to this optimization bailout. 

Only standalone instances of `process` or `global`.

### Files to fix

- [ ] https://github.com/OfficeDev/office-ui-fabric-react/blob/5c78f7e4c7a08715cdfde21981fd48c6d59c1e62/packages/utilities/src/object.ts#L9-L11

- [ ] https://github.com/OfficeDev/office-ui-fabric-react/blob/5c78f7e4c7a08715cdfde21981fd48c6d59c1e62/packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.base.tsx#L33-L37

- [ ] https://github.com/OfficeDev/office-ui-fabric-react/blob/5c78f7e4c7a08715cdfde21981fd48c6d59c1e62/packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.base.tsx#L33-L38

- [ ] https://github.com/OfficeDev/office-ui-fabric-react/blob/5c78f7e4c7a08715cdfde21981fd48c6d59c1e62/packages/office-ui-fabric-react/src/components/Dialog/Dialog.base.tsx#L44-L61

- [ ] https://github.com/OfficeDev/office-ui-fabric-react/blob/5c78f7e4c7a08715cdfde21981fd48c6d59c1e62/packages/office-ui-fabric-react/src/components/FocusZone/FocusZone.tsx#L94-L100

- [ ] https://github.com/OfficeDev/office-ui-fabric-react/blob/5c78f7e4c7a08715cdfde21981fd48c6d59c1e62/packages/office-ui-fabric-react/src/components/Layer/Layer.base.tsx#L41-L45

- [ ] https://github.com/OfficeDev/office-ui-fabric-react/blob/5c78f7e4c7a08715cdfde21981fd48c6d59c1e62/packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.base.tsx#L43-L55

- [ ] https://github.com/OfficeDev/office-ui-fabric-react/blob/5c78f7e4c7a08715cdfde21981fd48c6d59c1e62/packages/office-ui-fabric-react/src/components/TextField/TextField.base.tsx#L93-L104

- [ ] https://github.com/OfficeDev/office-ui-fabric-react/blob/5c78f7e4c7a08715cdfde21981fd48c6d59c1e62/packages/utilities/src/warn/warn.ts#L12-L16

- [ ] https://github.com/OfficeDev/office-ui-fabric-react/blob/5c78f7e4c7a08715cdfde21981fd48c6d59c1e62/packages/utilities/src/warn/warnConditionallyRequiredProps.ts#L19-L26

- [ ] https://github.com/OfficeDev/office-ui-fabric-react/blob/5c78f7e4c7a08715cdfde21981fd48c6d59c1e62/packages/utilities/src/warn/warnDeprecations.ts#L12-L23

- [ ] https://github.com/OfficeDev/office-ui-fabric-react/blob/5c78f7e4c7a08715cdfde21981fd48c6d59c1e62/packages/utilities/src/warn/warnMutuallyExclusive.ts#L11-L20

### Considerations

This was actually introduced to provide support for customers who are not bundling their applications with webpack/rollup. (AMD users). For the sake of performance here, those customers should pre bundle their code instead since it can provide treeshaking and AMD is heavily not-performant. 

Happy to work with teams here to address this issue. 

@dzearing @kenotron 

tl:dr: any customer today who uses a component like button, which then imports Dialog in a weird way, etc. will have a bunch of extra un-shakeable code.

**Comment by dzearing** [comment]
To clarify @TheLarkInn replace this:

```js
if (typeof process !== 'undefined' && process.env.NODE_ENV !== 'production') { 
```

With this:

```js
if (process.env.NODE_ENV !== 'production') { 
```

@ThomasMichon fyi

**Comment by apneer** [comment]
As discussed with Sean, he will make the change.
